pennylane/tape/qscript.py

```diff
@@ -1326,19 +1327,45 @@ def map_to_standard_wires(self) -> "QuantumScript":
         >>> circuit.map_to_standard_wires() is circuit
         True
 
+        Work wires that are not used in operations or measurements are mapped to the last
+        positions in the circuit:
+
+        >>> mcx = qml.MultiControlledX([1, 4, "b", 2], work_wires=[0, 6])
+        >>> circuit = qml.tape.QuantumScript([mcx], [qml.probs(wires=[2, 3, 6])])
+        >>> mapped_circuit = circuit.map_to_standard_wires()
+        >>> mapped_circuit.circuit
+        [MultiControlledX(wires=[0, 1, 2, 3], control_values=[True, True, True]),
+         probs(wires=[3, 4, 5])]
+        >>> mapped_circuit[0].work_wires
+        Wires([6, 5])
         """
+        wire_map = self._get_standard_wire_map()
+        if wire_map is None:
+            return self
+        tapes, fn = qml.map_wires(self, wire_map)
+        return fn(tapes)
+
+    def _get_standard_wire_map(self) -> dict:
+        """Helper function to produce the wire map for map_to_standard_wires."""
         op_wires = Wires.all_wires(op.wires for op in self.operations)
+        work_wires = Wires.all_wires(getattr(op, "work_wires", []) for op in self.operations)
         meas_wires = Wires.all_wires(mp.wires for mp in self.measurements)
         num_op_wires = len(op_wires)
         meas_only_wires = set(meas_wires) - set(op_wires)
-        if set(op_wires) == set(range(num_op_wires)) and meas_only_wires == set(
-            range(num_op_wires, num_op_wires + len(meas_only_wires))
+        num_op_meas_wires = num_op_wires + len(meas_only_wires)
+        work_only_wires = set(work_wires) - set(op_wires) - meas_only_wires
+        # If the op wires are consecutive integers, followed by measurement-only wires, followed
+        # by work-only wires, we do not perform a mapping, signalled by returning `None`.
+        if (
+            set(op_wires) == set(range(num_op_wires))
```

Comment:
Does the order of wires not matter here? Let's say if the op wires are `[3, 2, 1, 0]`, is it expected to be mapped to `[0, 1, 2, 3]`?

Reply:
No, it does not.
Interestingly, this is a function that does nothing on some inputs  (like you example), even though it _would_ change the wire ordering if one _did_ apply the core functionality.

The reason is that there is no canonical wire _ordering_ given a circuit, beyond the grouping into `op_wires`, `meas_only_wires`, and now `work_only_wires`. The main thing is to assign consecutive integers.